itest/rust/src/object_tests/call_deferred_test.rs

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+use crate::framework::itest;
+use godot::obj::WithBaseField;
+use godot::prelude::*;
+use godot::task::{SignalFuture, TaskHandle};
+use std::ops::DerefMut;
+
+const ACCEPTED_NAME: &str = "touched";
+
+#[derive(GodotClass)]
+#[class(init,base=Node2D)]
+struct DeferredTestNode {
+    base: Base<Node2D>,
+}
+
+#[godot_api]
+impl DeferredTestNode {
+    #[signal]
+    fn test_completed(name: StringName);
+
+    #[func]
+    fn gd_accept(&mut self) {
+        self.rust_accept();
+    }
+
+    fn rust_accept(&mut self) {
+        self.base_mut().set_name(ACCEPTED_NAME);
+    }
+
+    fn as_expectation_task(&self) -> TaskHandle {
```

Comment:
The `as_` convention is used for references.

And I'm not sure if "expectation" is really clear in this context -- maybe `to_testable_task`?

Reply:
Thanks for the pointer about `as`. testable implies to me something that can be tested, but not something that does the test.  Does `to_assertion_task` sound clearer to you? It is the task handle that asserts.

Reply:
Yeah, that's better :ok_hand: